client/components/svg-canvas.jsx

```diff
@@ -64,23 +96,70 @@ export default class SVGCanvas extends React.Component {
 
     // Put the new object back in the original location
     this.setState({
-      drawnPaths: [...this.state.drawnPaths.slice(0, drawnPathsIdx), newCurrentPath, ...this.state.drawnPaths.slice(drawnPathsIdx + 1)]
+      elements: [...this.state.elements.slice(0, elementsIdx), newCurrentPath, ...this.state.elements.slice(elementsIdx + 1)]
+    });
+  }
+
+  addUserInputToTextData(userInputKey, elementsIdx) {
+    // Check for valid characters BEFORE calling this method
+    const modifyTextObj = this.state.elements[elementsIdx];
+    let newTextData = modifyTextObj.userInput;
+    if (userInputKey === 'Backspace') {
+      newTextData = newTextData.slice(0, -1);
+    } else {
+      newTextData = newTextData + userInputKey;
+    }
+
+    const newCurrentText = {
+      elementType: modifyTextObj.elementType,
+      elementId: modifyTextObj.elementId,
+      startingPoint: modifyTextObj.startingPoint,
+      userInput: newTextData,
+      fill: modifyTextObj.fill,
+      fontSize: modifyTextObj.fontSize
+    };
```

Comment:
This is a very cosmetic tip, but you can "spread" object properties to create a shallow copy. I think there are a couple of places where this is applicable.

```js
const newCurrentText = {
  ...modifyTextObj,
  userInput: newTextData
};
```

Reply:
I didn't know that I could overwrite a property from a spread by setting it again afterwards. I'll definitely make this change.